message-markdown.md

```diff
@@ -0,0 +1,44 @@
+# Markdown in messages
+
+Deltachat is going to support the following subset of markdown:
+
+### `**bold**`
+
+### `__underline__`
+
+### `~~strikethrough~~`
+
+### `https://delta.chat` - Urls
+
+Make URLs clickable, but **don't** support the Link syntax `[Name](url)`
+because that could be used for misleading and as such be used in scams.
+
+### `` `inline-code` ``
+useful to send non markdown text in your message like source code snippets.
+Should get rendered in a monospace font and with a different background.
+ 
+### ` ``` fence code block ``` `
+
+```md
+Similar to `inline-code` but not inline and it may supports code highlighting.
+```
+` ```[lang?] [content]``` `
+ A bit modified from the common syntax to allow one liners.
+
+### `:emoji:`
+
+## Future:
+
+### `mailto:email@address.example.com`
+
+Make mailto links clickable with all parameters: `?subject=Sample%20Subject&body=Sample%20Body`
+
+### Custom Deltachat URI Scheme
+see https://support.delta.chat/t/custom-deltachat-url-scheme/346
+
+### Mentions `@username`
+Clickable. (could get replaced with an user hash/email/id on send/on recieve so that it's still valid on name change.)
```

Comment:
yeah, that would be nice. however, it might get a bit tricky as there is not really a unique username currently. @user@example.org looks a bit strange.

Reply:
The user will probably type the current username, but dc can save/transmit it in another format.
Discord has an even more strange code https://blog.discordapp.com/how-discord-renders-rich-messages-on-the-android-app-67b0e5d56fbe

Reply:
>  however, it might get a bit tricky as there is not really a unique username currently.

one idea is that as simon says, when mentioning the original email contains the user's email, just that dc replaces/show that email in messages as a @tag with the display name we have for that user, if we don't know that address the full email should be shown then, it is just a display thing, internally no name is sent, only email addresses which should be unique